lasagne/layers/sparse/noise.py

```diff
@@ -0,0 +1,40 @@
+import theano
+import theano.sparse as sparse
+
+from ..noise import DropoutLayer as Dropout
+
+
+__all__ = [
+    "DropoutLayer",
+    "dropout",
+]
+
+
+class DropoutLayer(Dropout):
+    def get_output_for(self, input, deterministic=False, **kwargs):
+        """
+        Parameters
+        ----------
+        input : theano sparse matrix
+            output from the previous layer
+        deterministic : bool
+            If true dropout and scaling is disabled, see notes
+        """
+        if deterministic or self.p == 0:
+            return input
+        else:
+            assert type(input) == sparse.basic.SparseVariable
+
+            retain_prob = 1 - self.p
+            if self.rescale:
+                input = sparse.basic.mul(input, 1/retain_prob)
```

Comment:
@f0k 
I don't know the reason, but after adding `lasagne.utils.floatX`, my CPU computation increased dramatically. In my experiment, I have a `csr_matrix` dataset with `shape == (728497, 53097)`.

Before including `lasagne.utils.floatX`, my process was using only one core, and each epoch was taking around 146 seconds. After adding it, all the 16 cores were getting considerably busy and each epoch was taking around 463 seconds. Maybe this is the reason why `layers.DropoutLayer.get_output_for()` does NOT use `lasagne.utils.floatX`.

So, I am going to revert this change. I hope that it's ok for you.